src/Avalonia.Base/Metadata/InheritDataTypeFromAttribute.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+
+namespace Avalonia.Metadata;
+
+public enum InheritDataTypeFromScopeKind
```

Comment:
Nit: this may need a comment to indicate that `InheritDataTypeFromScopeKind` and `ScopeTypes` must have equivalent values, so we don't inadvertently change only one of the two enums in the future.

Reply:
Replaced with a switch.